server/controllers/reports_proposed/data/variation_exploitation.js

```diff
@@ -0,0 +1,174 @@
+    parent.balanceCurrent = result.balance_curent;
+    parent.viewBalanceCurrent = getNumeralFormat(parent.balanceCurrent);
+    parent.variationValue = getNumeralFormat(parent.balanceCurrent - parent.balancePrevious);
+    finalList.push(parent);
+  });
+
+  return finalList;
+}
+
+function getParentAccount(parentId){
+  var found;
+  for (var i = currentExploitations.length - 1; i >= 0; i--) {
+    if(currentExploitations[i].id == parentId){
+      found = currentExploitations[i];
+      break;
+    }
+  };
+  return found;
+}
+
+function getSubAccountsOnly (list){
+  return list.filter(function (item){
+    return item.account_type_id === 1;
+  });
+}
+
+function getAllChild (parentId){
+  var founds = [], cTotalDebit = 0,
+      cTotalCredit = 0, cTotalBalance = 0,
+      pTotalDebit = 0, pTotalCredit = 0, pTotalBalance = 0, inds_c = [], inds_p = [];
+
+  for (var i = currentSubAccounts.length - 1; i >= 0; i--) {
+    if(currentSubAccounts[i].parent === parentId){
+      currentSubAccounts[i].balance = (currentSubAccounts[i].balance) ? currentSubAccounts[i].balance : 0;
+      currentSubAccounts[i].viewBalance = getNumeralFormat(currentSubAccounts[i].balance);
+
+      currentSubAccounts[i].previousBalance = getPreviousState(currentSubAccounts[i].id).balance;
+      currentSubAccounts[i].viewPreviousBalance = getNumeralFormat(currentSubAccounts[i].previousBalance);
+
+      currentSubAccounts[i].percent = getPercent(currentSubAccounts[i].balance - currentSubAccounts[i].previousBalance, currentSubAccounts[i].previousBalance);
+      currentSubAccounts[i].variationValue = getNumeralFormat(currentSubAccounts[i].balance - currentSubAccounts[i].previousBalance);
+
+      founds.push(currentSubAccounts[i]);
+
+      cTotalBalance +=currentSubAccounts[i].balance ? currentSubAccounts[i].balance : 0;
+      pTotalBalance +=currentSubAccounts[i].previousBalance ? currentSubAccounts[i].previousBalance : 0;
+      inds_c.push(i);          
+    }
+  };
+
+  for (var i = previousSubAccounts.length - 1; i >= 0; i--) {
+    if(previousSubAccounts[i].parent === parentId){
+      previousSubAccounts[i].previousBalance = (previousSubAccounts[i].balance) ? previousSubAccounts[i].balance : 0;
+      previousSubAccounts[i].viewPreviousBalance = getNumeralFormat(previousSubAccounts[i].balance);
+
+      previousSubAccounts[i].balance = 0;
+      previousSubAccounts[i].viewBalance = getNumeralFormat(previousSubAccounts[i].balance);
+
+      previousSubAccounts[i].percent = getPercent(previousSubAccounts[i].balance - previousSubAccounts[i].previousBalance, previousSubAccounts[i].previousBalance);
+      previousSubAccounts[i].variationValue = getNumeralFormat(previousSubAccounts[i].balance - previousSubAccounts[i].previousBalance);
+
+      founds.push(previousSubAccounts[i]);
+      pTotalBalance +=previousSubAccounts[i].previousBalance ? previousSubAccounts[i].previousBalance : 0;
+      inds_p.push(i);
+    }
+  };
+
+  inds_c.forEach(function (item){
```

Comment:
It is not at all clear what you are doing here.  What are you doing?  Can you provide some comments?